google/resource_compute_disk.go

```diff
@@ -390,213 +747,359 @@ func resourceComputeDiskDelete(d *schema.ResourceData, meta interface{}) error {
 			}
 		}
 	}
+	log.Printf("[DEBUG] Deleting Disk %q", d.Id())
+	res, err := Delete(config, url)
+	if err != nil {
+		return fmt.Errorf("Error deleting Disk %q: %s", d.Id(), err)
+	}
 
-	// Delete the disk
-	op, err := config.clientCompute.Disks.Delete(
-		project, z, d.Id()).Do()
+	op := &compute.Operation{}
+	err = Convert(res, op)
 	if err != nil {
-		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
-			log.Printf("[WARN] Removing Disk %q because it's gone", d.Get("name").(string))
-			// The resource doesn't exist anymore
-			d.SetId("")
-			return nil
-		}
-		return fmt.Errorf("Error deleting disk: %s", err)
+		return err
 	}
 
-	err = computeOperationWaitTime(config.clientCompute, op, project, "Deleting Disk", int(d.Timeout(schema.TimeoutDelete).Minutes()))
+	err = computeOperationWaitTime(
+		config.clientCompute, op, project, "Deleting Disk",
+		int(d.Timeout(schema.TimeoutDelete).Minutes()))
+
 	if err != nil {
 		return err
 	}
 
-	d.SetId("")
+	log.Printf("[DEBUG] Finished deleting Disk %q: %#v", d.Id(), res)
 	return nil
 }
 
-// Is the new disk size smaller than the old one?
-func isDiskShrinkage(old, new, _ interface{}) bool {
-	// It's okay to remove size entirely.
-	if old == nil || new == nil {
-		return false
+func resourceComputeDiskImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	config := meta.(*Config)
+	parseImportId([]string{"projects/(?P<project>[^/]+)/zones/(?P<zone>[^/]+)/disks/(?P<name>[^/]+)", "(?P<project>[^/]+)/(?P<zone>[^/]+)/(?P<name>[^/]+)", "(?P<name>[^/]+)"}, d, config)
+
+	// Replace import id for the resource id
+	id, err := replaceVars(d, config, "{{name}}")
+	if err != nil {
+		return nil, fmt.Errorf("Error constructing id: %s", err)
 	}
-	return new.(int) < old.(int)
-}
+	d.SetId(id)
+	if zone, err := getZone(d, config); err != nil || zone == "" {
```

Comment:
I wouldn't mind a comment in here that explains what's going on

Reply:
Man, me either.  I had to figure it out all over again.  Done.